File: backend/app/schemas/readiness.py
from pydantic import BaseModel


class QualityEstimateOut(BaseModel):
    score: float
    judge_component: float
    judge_precision: float
    ci_low: float
    ci_high: float
    observation_count: int
    vibe_check_size: int


class EvidenceCoverageOut(BaseModel):
    # `annotated_*` (saved Annotations on Lab docs) and
    # `field_evidence_*` (per-field quote evidence) intentionally collide on
    # the same surface, so naming them by what they actually count avoids
    # the dogfood-#7 confusion of "72 fields reviewed" vs "0% with field
    # evidence".
    annotated_docs: int
    annotated_entities: int
    annotated_fields: int
    field_evidence_fields: int
    field_evidence_coverage_ratio: float


class SchemaMaturityOut(BaseModel):
    # draft | stabilizing | lock_candidate | locked
    status: str
    annotated_docs: int
    annotated_entities: int
    recent_schema_breaking_changes: int
    message: str


class RegressionHealthOut(BaseModel):
    counterexamples_total: int
    counterexample_component: float | None
    # no_production_feedback | passing | failing | unknown
    status: str


class RiskyFieldOut(BaseModel):
    field_name: str
    count: int


class APIReadinessOut(BaseModel):
    quality_estimate: QualityEstimateOut
    evidence_coverage: EvidenceCoverageOut
    schema_maturity: SchemaMaturityOut
    regression_health: RegressionHealthOut
    risky_fields: list[RiskyFieldOut]
    publish_blockers: list[str]
    warnings: list[str]

File: frontend/src/types/readiness.ts
// Mirror of backend/app/schemas/readiness.py — keep field names and types
// in sync. Status string unions match the discrete values produced by
// backend/app/services/readiness.py (build_readiness).

export type SchemaMaturityStatus =
  | "draft"
  | "stabilizing"
  | "lock_candidate"
  | "locked";

export type RegressionHealthStatus =
  | "no_production_feedback"
  | "passing"
  | "failing"
  | "unknown";

export type QualityEstimate = {
  score: number;
  judge_component: number;
  judge_precision: number;
  ci_low: number;
  ci_high: number;
  observation_count: number;
  vibe_check_size: number;
};

export type EvidenceCoverage = {
  // Dogfood follow-up #7: `annotated_*` (saved Annotations on Lab docs)
  // and `field_evidence_*` (per-field quote evidence) used to collide on
  // the same surface as "reviewed". Names now match what they actually
  // count.
  annotated_docs: number;
  annotated_entities: number;
  annotated_fields: number;
  field_evidence_fields: number;
  field_evidence_coverage_ratio: number;
};

export type SchemaMaturity = {
  status: SchemaMaturityStatus;
  annotated_docs: number;
  annotated_entities: number;
  recent_schema_breaking_changes: number;
  message: string;
};

export type RegressionHealth = {
  counterexamples_total: number;
  counterexample_component: number | null;
  status: RegressionHealthStatus;
};

export type RiskyField = {
  field_name: string;
  count: number;
};

export type APIReadinessOut = {
  quality_estimate: QualityEstimate;
  evidence_coverage: EvidenceCoverage;
  schema_maturity: SchemaMaturity;
  regression_health: RegressionHealth;
  risky_fields: RiskyField[];
  publish_blockers: string[];
  warnings: string[];
};
